frame/staking/src/scheduler.rs

```diff
@@ -0,0 +1,294 @@
+// TODO: move this to frame/support/scheduler.rs
```

Comment:
maybe we can merge this into pallet-scheduler?

Reply:
moving it to `frame-support/scheduler`, probably but to the pallet, I don't think so 🤔 

The scheduler pallet is in the business of scheduling `Calls` and deals with Origin and such. This little PoC is about scheduling trusted code inside the runtime. Perhaps with some stretching, we could implement this with the scheduler pallet altogether, but end of the day I think they are doing different things.